apache2/modsecurity.c

```diff
@@ -154,85 +154,37 @@ int modsecurity_init(msc_engine *msce, apr_pool_t *mp) {
     curl_global_init(CURL_GLOBAL_ALL);
 #endif
     /* Serial audit log mutext */
-    rc = apr_global_mutex_create(&msce->auditlog_lock, NULL, APR_LOCK_DEFAULT, mp);
-    if (rc != APR_SUCCESS) {
+    msce->auditlog_lock = apr_pcalloc(mp, sizeof(struct waf_lock));
+    rc = Waf_createLock(msce->auditlog_lock, "auditlog_lock", strlen("auditlog_lock"), msc_waf_lock_owner);    
+    if (Waf_lock_isError(rc)) {
         //ap_log_error(APLOG_MARK, APLOG_ERR, rv, s, "mod_security: Could not create modsec_auditlog_lock");
         //return HTTP_INTERNAL_SERVER_ERROR;
         return -1;
     }
 
 #ifdef WAF_JSON_LOGGING_ENABLE
     /* Serial wafjson log mutext */
-    rc = apr_global_mutex_create(&msce->wafjsonlog_lock, NULL, APR_LOCK_DEFAULT, mp);
-    if (rc != APR_SUCCESS) {
+    msce->wafjsonlog_lock = apr_pcalloc(mp, sizeof(struct waf_lock));
+    rc = Waf_createLock(msce->wafjsonlog_lock, "wafjsonlog_lock", strlen("wafjsonlog_lock"), msc_waf_lock_owner);    
```

Comment:
just for consistentcy can we do `waf_jsonlog_lock`? The prefix `waf_` implying the component its being used with.

Reply:
ok